From 49fd486c23d61171e9202d6934167a89938f8a04 Mon Sep 17 00:00:00 2001
From: jules <jules@rawmaterialsoftware.com>
Date: Sat, 15 Jun 2013 16:53:47 +0100
Subject: [PATCH] Removed Graphics::drawTextAsPath, which is obsolete, since
 the same job can be done using Graphics::addTransform and any of the other
 drawText methods.

---
 modules/juce_graphics/contexts/juce_GraphicsContext.cpp | 10 ----------
 modules/juce_graphics/contexts/juce_GraphicsContext.h   | 13 +------------
 2 files changed, 1 insertion(+), 22 deletions(-)

diff --git a/modules/juce_graphics/contexts/juce_GraphicsContext.cpp b/modules/juce_graphics/contexts/juce_GraphicsContext.cpp
index 6af701c..0577507 100644
--- a/modules/juce_graphics/contexts/juce_GraphicsContext.cpp
+++ b/modules/juce_graphics/contexts/juce_GraphicsContext.cpp
@@ -251,16 +251,6 @@ void Graphics::drawSingleLineText (const String& text, const int startX, const i
     }
 }
 
-void Graphics::drawTextAsPath (const String& text, const AffineTransform& transform) const
-{
-    if (text.isNotEmpty())
-    {
-        GlyphArrangement arr;
-        arr.addLineOfText (context.getFont(), text, 0.0f, 0.0f);
-        arr.draw (*this, transform);
-    }
-}
-
 void Graphics::drawMultiLineText (const String& text, const int startX,
                                   const int baselineY, const int maximumLineWidth) const
 {
diff --git a/modules/juce_graphics/contexts/juce_GraphicsContext.h b/modules/juce_graphics/contexts/juce_GraphicsContext.h
index b290db4..816afc4 100644
--- a/modules/juce_graphics/contexts/juce_GraphicsContext.h
+++ b/modules/juce_graphics/contexts/juce_GraphicsContext.h
@@ -116,7 +116,7 @@ class JUCE_API  Graphics
         Note there's also a setFont (float, int) method to quickly change the size and
         style of the current font.
 
-        @see drawSingleLineText, drawMultiLineText, drawTextAsPath, drawText, drawFittedText
+        @see drawSingleLineText, drawMultiLineText, drawText, drawFittedText
     */
     void setFont (const Font& newFont);
 
@@ -158,17 +158,6 @@ class JUCE_API  Graphics
                             int startX, int baselineY,
                             int maximumLineWidth) const;
 
-    /** Renders a string of text as a vector path.
-
-        This allows a string to be transformed with an arbitrary AffineTransform and
-        rendered using the current colour/brush. It's much slower than the normal text methods
-        but more accurate.
-
-        @see setFont
-    */
-    void drawTextAsPath (const String& text,
-                         const AffineTransform& transform) const;
-
     /** Draws a line of text within a specified rectangle.
 
         The text will be positioned within the rectangle based on the justification
-- 
1.8.1.6

